### PR TITLE
add_TGBot_project_start

### PR DESCRIPTION
**Pull request by PavelBoPavel**
реализовано:
 - прохождение по кнопкам месяца, дней, часов, описание, контакты

### CODE REVIEW

**Review by Puzzletania — APPROVED** [review]

**Review by VasiliZas — APPROVED** [review]

**Review by yaroslav-zh — APPROVED** [review]

**Review by Puzzletania — APPROVED** [review]

File: src/test/java/com/github/pavelbo/registrationbot/FitbotApplicationTests.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.github.pavelbo.registrationbot.service.TelegramBot;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest(classes = {TelegramBot.class, ObjectMapper.class, MessageService.class})
+public class FitbotApplicationTests {
+
+	@Test
+	void contextLoads() {
+	}
```
Comment (KaydunovDenis) [inline src/test/java/com/github/pavelbo/registrationbot/FitbotApplicationTests.java:14]: I guess is don't must by empty.

File: src/test/java/com/github/pavelbo/registrationbot/service/TelegramBotTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+@SpringBootTest(classes = {TelegramBot.class, Mapper.class})
+class TelegramBotTest {
+
+    @Autowired
+    TelegramBot telegramBot;
+    @Autowired
+    ObjectMapper objectMapper;
+
+//    @Test
```
Comment (KaydunovDenis) [inline src/test/java/com/github/pavelbo/registrationbot/service/TelegramBotTest.java:22]: Почему закоментированно?

File: src/main/java/com/github/pavelbo/registrationbot/util/UpdateTypeUtil.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            } else if (update.getCallbackQuery().getData().contains("/hour")) {
+                return UpdateType.CALLBACK_HOUR;
+            } else {
+                return UpdateType.CALLBACK_ANOTHER;
+            }
+        } else if (isContactMessage(update)) {
+            return UpdateType.CONTACT;
+        } else {
+            return UpdateType.DEFAULT;
+        }
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/util/UpdateTypeUtil.java:25]: I think it need separete or rewrite on swith.

File: src/main/java/com/github/pavelbo/registrationbot/keyboard/ReplayKeyboardService.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class ReplayKeyboardService {
+    public ReplyKeyboardMarkup getReplayKeyboardMain() {
+        ReplyKeyboardMarkup keyBoardMarkup = new ReplyKeyboardMarkup();
+        List<KeyboardRow> keyboard = new ArrayList<>();
+        KeyboardRow row = new KeyboardRow();
+        KeyboardRow row2 = new KeyboardRow();
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/keyboard/ReplayKeyboardService.java:17]: I think we can use "var" here

File: src/main/java/com/github/pavelbo/registrationbot/keyboard/ReplayKeyboardService.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    public ReplyKeyboardMarkup getContact() {
+        ReplyKeyboardMarkup keyBoardMarkup = new ReplyKeyboardMarkup();
+        List<KeyboardRow> keyboard = new ArrayList<>();
+        KeyboardRow row = new KeyboardRow();
+        KeyboardButton keyboardButton = new KeyboardButton();
+        keyboardButton.setText("Отправить контакт!").setRequestContact(true);
+        row.add(keyboardButton);
+        keyboard.add(row);
+        keyBoardMarkup.setKeyboard(keyboard);
+        keyBoardMarkup.setResizeKeyboard(true);
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/keyboard/ReplayKeyboardService.java:37]: This code duplicate part code from the methode above.
I think can extract method here.

File: src/main/java/com/github/pavelbo/registrationbot/keyboard/ReplayKeyboardService.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    public ReplyKeyboardMarkup getYesOrNo() {
+        ReplyKeyboardMarkup keyBoardMarkup = new ReplyKeyboardMarkup();
+        List<KeyboardRow> keyboard = new ArrayList<>();
+        KeyboardRow row = new KeyboardRow();
+        row.add("Подтвердить!");
+        row.add("Отмена!");
+        keyboard.add(row);
+        keyBoardMarkup.setKeyboard(keyboard);
+        keyBoardMarkup.setResizeKeyboard(true);
+        return keyBoardMarkup;
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/keyboard/ReplayKeyboardService.java:50]: This code duplicate part code from the methode above.
I think can extract method here

File: src/main/java/com/github/pavelbo/registrationbot/service/CallbackAnotherProcessor.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+public class CallbackAnotherProcessor extends BaseProcessor implements Processor {
+
+    public void process(Update update) {
+        switch (update.getCallbackQuery().getData()) {
+            case "/start":
+                messageService.sendMessage(getChatId(update), START_MESSAGE,
+                        inlineKeyboardSource.getKeyboardMain(), replayKeyboardService.getReplayKeyboardMain());
+                break;
+            case "/contact":
+                messageService.sendMessage(getChatId(update), "Contact!", inlineKeyboardSource.getKeyboardMain());
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/service/CallbackAnotherProcessor.java:19]: Best practic create String constant for it.

File: src/main/java/com/github/pavelbo/registrationbot/service/CallbackAnotherProcessor.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+                break;
+            default:
+                messageService.sendMessage(getChatId(update), "Do no!", inlineKeyboardSource.getKeyboardMain(), replayKeyboardService.getReplayKeyboardMain());
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/service/CallbackAnotherProcessor.java:22]: Best practic create String constant for it.

File: src/main/java/com/github/pavelbo/registrationbot/service/CallbackHourProcessor.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+            case "/hour1800":
+            case "/hour1830":
+            case "/hour1900":
+            case "/hour1930":
+            case "/hour2000":
+            case "/hour2030":
+            case "/hour2100":
+            case "/hour2130":
+            case "/hour2200":
+                //saveFile(update);
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/service/CallbackHourProcessor.java:41]: You can to write
//TODO come doing, realise new function here.... etc

File: src/main/java/com/github/pavelbo/registrationbot/service/ContactProcessor.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.stereotype.Service;
+import org.telegram.telegrambots.meta.api.objects.Update;
+
+import static com.github.pavelbo.registrationbot.util.ChatIdExtractor.getChatId;
+
+@Service
+public class ContactProcessor extends BaseProcessor implements Processor{
+
+    public void process(Update update) {
+        messageService.sendMessage(getChatId(update),"Авторизация прошла успешно!", replayKeyboardService.getReplayKeyboardMain());
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/service/ContactProcessor.java:12]: Best practic create String constant for it.

File: src/main/java/com/github/pavelbo/registrationbot/service/DefaultProcessor.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.stereotype.Service;
+import org.telegram.telegrambots.meta.api.objects.Update;
+
+import static com.github.pavelbo.registrationbot.util.ChatIdExtractor.getChatId;
+
+@Service
+public class DefaultProcessor extends BaseProcessor implements Processor{
+
+    public void process(Update update) {
+        messageService.sendMessage(getChatId(update),"Неизвестная для бота команда!", replayKeyboardService.getReplayKeyboardMain());
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/service/DefaultProcessor.java:12]: Best practic create String constant for it.

File: src/main/java/com/github/pavelbo/registrationbot/util/ChatIdExtractor.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+    public static Long getChatId(Update update) {
+        if (isTextMessage(update))
+            return update.getMessage().getChatId();
+        else if (isCallbackQuery(update))
+            return update.getCallbackQuery().getMessage().getChatId();
+        else if (update.getMessage().hasContact()){
+            return update.getMessage().getChatId();
+        }
+        return null;
```
Comment (KaydunovDenis) [inline src/main/java/com/github/pavelbo/registrationbot/util/ChatIdExtractor.java:18]: The null can throw NullPointException.
It is not good practic.
I think we need to throw the error or handle it elsewhere.

**Review by KaydunovDenis — CHANGES_REQUESTED** [review]
Good gob!
This is hard for me. If you can, create a smaller pull request next time.
30 files  takes a lot of time.
I neen more time for good review.
See my comment. I did several remark.

**Review by KaydunovDenis — APPROVED** [review]

**Comment by KaydunovDenis** [comment]
Мы с этим что-то будем делать? может как то смержим? А потом уже будем его изменять?

**Pull request closed without merging**